File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3031_minimum-time-to-revert-word-to-initial-state-ii/solutions/java/solution.java
class Solution {
    public int minimumTimeToInitialState(String word, int k) {
        int n = word.length();
        int g = gcd(n, k);
        return n / g;
    }
    
    private int gcd(int a, int b) {
        while (b != 0) {
            int t = b;
            b = a % b;
            a = t;
        }
        return a;
    }
}

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3031_minimum-time-to-revert-word-to-initial-state-ii/solutions/python3/solution.py
class Solution:
    def minimumTimeToInitialState(self, word: str, k: int) -> int:
        n = len(word)
        if k == n:
            return 1
        
        # Calculate the cycle length to revert back to the initial position
        cycle_length = n // k
        if n % k == 0:
            return cycle_length
        else:
            return cycle_length + 1
